programs/CMakeLists.txt

```diff
@@ -446,13 +453,17 @@ else ()
         install (FILES "${CMAKE_CURRENT_BINARY_DIR}/clickhouse-static-files-disk-uploader" DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT clickhouse)
         list(APPEND CLICKHOUSE_BUNDLE clickhouse-static-files-disk-uploader)
     endif ()
+
     if (ENABLE_CLICKHOUSE_KEEPER)
-        if (NOT BUILD_STANDALONE_KEEPER)
+        if (NOT BUILD_STANDALONE_KEEPER AND CREATE_KEEPER_SYMLINK)
             add_custom_target (clickhouse-keeper ALL COMMAND ${CMAKE_COMMAND} -E create_symlink clickhouse clickhouse-keeper DEPENDS clickhouse)
             install (FILES "${CMAKE_CURRENT_BINARY_DIR}/clickhouse-keeper" DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT clickhouse)
         endif()
 
-        list(APPEND CLICKHOUSE_BUNDLE clickhouse-keeper)
+        # otherwise we don't build keeper
+        if (BUILD_STANDALONE_KEEPER OR CREATE_KEEPER_SYMLINK)
+            list(APPEND CLICKHOUSE_BUNDLE clickhouse-keeper)
+        endif()
```

Comment:
It looks redundant to me. We need define ENABLE_CLICKHOUSE_KEEPER as `BUILD_STANDALONE_KEEPER OR CREATE_KEEPER_SYMLINK` above